runtime/lua/vim/lsp/util.lua

```diff
@@ -226,6 +226,165 @@ end
 -- function M.glob_to_regex(glob)
 -- end
 
+--@private
+--- Finds the first line and column of the difference between old and new lines
+--@param old_lines table list of lines
+--@param new_lines table list of lines
+--@returns (int, int) start_line_idx and start_col_idx of range
+local function first_difference(old_lines, new_lines)
+	local line_count = math.min(#old_lines, #new_lines)
+	if line_count == 0 then return 1, 1 end
+	local start_line_idx = line_count
+	for i = 1, line_count do
+		if old_lines[i] ~= new_lines[i] then
+			start_line_idx = i
+			break
+		end
+	end
+	if start_line_idx >= line_count then
+		return line_count, #old_lines[line_count]
```

Comment:
What if the change happens on line `line_count` but before the last char of the line? That would result in a wrong column index.

Perhaps `start_line_idx` should be initialized with `line_count+1` or another value not in the range 1..`line_count` to 
take care of this.